libmamba/include/mamba/util/flat_set.hpp

```diff
@@ -565,7 +565,7 @@ struct std::hash<mamba::util::flat_set<Key, Compare, Allocator>>
 {
     auto operator()(const mamba::util::flat_set<Key, Compare, Allocator>& set) const -> std::size_t
     {
-        return mamba::util::hash_vals(set);
+        return mamba::util::hash_range(set);
```

Comment:
Why this change?

Reply:
`hash_vals` hashes `set` as a value rather than its' internal value (creating an infinite recursion).

`hash_range` hashes elements considers `set` a range and hashes all of its elements.

Reply:
I'm actually wondering about the use cases of using `hash_vals`, and btw I just realized there are no tests for `hash_vals`. Would you mind adding them please? (in this PR or another one, as you prefer).

Reply:
`hash_vals` is used to combine the hashes of individual scalar values. I can open a PR after this one to test its behavior.